3.6-rc/alpine/docker-entrypoint.sh

```diff
@@ -88,6 +80,34 @@ declare -A configDefaults=(
 	[ssl_verify]='verify_peer'
 )
 
+# allow the container to be started with `--user`
+if [[ "$1" == rabbitmq* ]] && [ "$(id -u)" = '0' ]; then
+	# this needs to happen late enough that we have the SSL config
+	# https://github.com/docker-library/rabbitmq/issues/283
+	for conf in "${allConfigKeys[@]}"; do
+		var="RABBITMQ_${conf^^}"
+		val="${!var:-}"
+		[ -n "$val" ] || continue
+		case "$conf" in
+			*_ssl_*file | ssl_*file )
+				if [ -f "$val" ]; then
```

Comment:
i.e., it could already be owned by user `999` or be accessible by a group that we are in (`--group-add`) or even world readable. :astonished: As far as I can tell from the documentation, it shouldn't care how it can read it as long as it can (and access the directory): "have the appropriate permissions".

1. https://www.rabbitmq.com/ssl.html#enabling-tls-paths
1. https://www.rabbitmq.com/troubleshooting-ssl.html#verify-file-permissions

Reply:
Really good point -- we should verify access and throw a warning too.

Reply:
Updated!